services/degiro-auth/main.py

```
@@ -0,0 +1,300 @@
+    blob = json.dumps({"sessionId": result["sessionId"], "intAccount": int_account})
+    return {"sessionBlob": blob}
+
+
+@app.post("/portfolio")
+async def portfolio(req: PortfolioRequest):
+    try:
+        parsed = json.loads(req.sessionBlob)
+        session_id = parsed["sessionId"]
+        int_account = parsed["intAccount"]
+    except (json.JSONDecodeError, KeyError):
+        raise HTTPException(status_code=400, detail="Invalid sessionBlob format")
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)